[panomia01] iP

panomia01 opened the pull request:
# Dew: Enhanced Task Management

**Dew** takes the stress out of managing your tasks and keeps you focused on what matters. It's:
- **Text-based** for simplicity
- **Easy to learn** for quick onboarding
- **FAST** and **SUPER FAST** for task management

## Features
- [x] Managing tasks
- [x] Searching for tasks by keywords (New feature 🎉)
- [x] Managing deadlines
- [x] Marking and unmarking tasks as complete
- [x] Deleting tasks
- [x] Viewing all tasks in a list

---

## How to Use Dew
1. **Download** the app from [here](#).
2. **Run** the application via:
   ```bash
   java -jar Dew.jar

## 
If you're a Java programmer, you can also use **Dew** to practice your skills. Check out the main entry point below:

```java
public class Main {
    public static void main(String[] args) {
        new Dew("src/main/tasks.txt").run();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

panomia01 committed:
Change Duke to Dew and add start and end dialogue

panomia01 committed:
Added user input

panomia01 committed:
Added additional lines for end dialogue

panomia01 committed:
added list to show user input

panomia01 committed:
added different tier of tasks

panomia01 committed:
Test using the I/O redirection technique

panomia01 committed:
level 5 and 6 completed

panomia01 committed:
Add save function into dew

panomia01 committed:
Add pre-load tasks

panomia01 committed:
Change date and time format

DiligentPenguinn commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
```
Perhaps a noun would be more intuitive

DiligentPenguinn commented on src/main/java/Dew.java:
```diff
@@ -0,0 +1,180 @@
+                    + "|____/ \\___|  \\__/   \\__/\n";
+        System.out.println("Hello from\n" + logo);
+
+        String start_dialogue = "____________________________________________________________\n" +
```
```suggestion
        String startDialogue = "____________________________________________________________\n" +
```

DiligentPenguinn commented on src/main/java/Dew.java:
```diff
@@ -0,0 +1,180 @@
+                " What can I do for you?\n" +
+                "____________________________________________________________\n";
+
+        String end_dialogue = "____________________________________________________________\n" +
```
```suggestion
        String endDialogue = "____________________________________________________________\n" +
```

DiligentPenguinn commented on src/main/java/Dew.java:
```diff
@@ -0,0 +1,180 @@
+
+    private static void loadTasks(ArrayList<Task> items) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(FILE_PATH))) {
+            String line;
```
Do you think `line` is a bit ambiguous?

DiligentPenguinn commented on src/main/java/Dew.java:
```diff
@@ -0,0 +1,180 @@
+                            + "   " + items.get(itemIndex) + "\n"
+                            + line);
+                } else if (input.equals("todo")) {
+                    String desc = scanner.nextLine();
```
Do you think `description` would be more consistent with the rest? (I saw that in other parts you named it `description` instead)

DiligentPenguinn reviewed (commented):
Overall, I found your code quality good, except a few naming inconsistencies.

panomia01 committed:
Add new OOP

panomia01 committed:
Change imports to packages

panomia01 committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

panomia01 committed:
Add gradle into project

panomia01 committed:
Add J-unit for testing purpose

panomia01 committed:
Create jar file

panomia01 committed:
Add javadocs comments

panomia01 committed:
Tweak the code to comply with a coding standard

panomia01 committed:
Add find feature

panomia01 committed:
Add additional tasks for testing

HarishB99 commented on src/main/java/dew/DewException.java:
```diff
@@ -0,0 +1,18 @@
+     */
+    public DewException(String message) {
+        super("____________________________________________________________\n"
+                + message + "\n"
```
```suggestion
                + message + System.lineSeparator()
```

HarishB99 commented on src/main/java/dew/DewException.java:
```diff
@@ -0,0 +1,18 @@
+     * @param message The error message to be displayed.
+     */
+    public DewException(String message) {
+        super("____________________________________________________________\n"
```
```suggestion
        super("____________________________________________________________" + System.lineSeparator()
```

HarishB99 commented on src/main/java/dew/DewException.java:
```diff
@@ -0,0 +1,18 @@
+    public DewException(String message) {
+        super("____________________________________________________________\n"
+                + message + "\n"
+                + "_________________________________________________________________\n");
```
```suggestion
                + "_________________________________________________________________" + System.lineSeparator());
```

HarishB99 commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.getMonth() + " " + by.getDayOfMonth() + " " + by.getYear() + ")";
```
Maybe this line can be broken down into two?

HarishB99 reviewed (commented):
1. There were inconsistencies (but very few) in the way the import statements were grouped (e.g., Parser.java and Storage.java).
2. Maybe you can try replacing "\n" with System.lineSeparator()?
3. I noticed dewTest.java contains dummy test cases. Will you be adding more test cases in the future?

In overall, your code looked very organised, clean and was easy to read and follow. I liked how your test cases were well formatted with descriptions for each test case. I also liked how you made use of `@BeforeEach` annotation to setup the necessary variables for your test cases. Good job!

panomia01 committed:
Add checkstyle function

panomia01 committed:
Add GUI function

panomia01 committed:
Add assertions to improve code robustness

Assertions in key areas helps to catch invalid states early during
development.

Add assertions in key areas to catch invalid states early:
- Ensure `userInput` is not null or empty in `Parser.java`
before parsing.
- Validate that the file path is not empty and tasks exist in
`Storage.java`.
- Check that messages are not null before displaying them in `Ui.java`.
- Verify that the `dew` instance is initialized in `Main.java`.

Assertions help detect issues during development but remain disabled in
production, ensuring reliability without runtime overhead.

panomia01 committed:
Edit to adjust code quality

panomia01 committed:
Merge pull request #2 from panomia01/branch-A-Assertions

Add assertions to improve code robustness

panomia01 committed:
Merge pull request #3 from panomia01/branch-A-CodeQuality

Edit to adjust code quality

panomia01 committed:
Add Streams to Parser

panomia01 committed:
Merge pull request #4 from panomia01/branch-A-Streams

Add Streams to Parser

panomia01 committed:
Add update functionality to edit each entry

panomia01 committed:
Add user guide and UI photo

panomia01 committed:
Fix bugs and add background

panomia01 committed:
Fix some bugs

panomia01 committed:
Fix some bugs in parser and improve GUI

panomia01 committed:
Fix storage bugs

panomia01 committed:
Fix readme docs and add title

panomia01 committed:
Edit readme doc

panomia01 committed:
Improve date and time feature and exception handling

panomia01 committed:
Edit readme for better reference

panomia01 committed:
Edit deadline timing

panomia01 committed:
Fix some command bugs
